src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -569,6 +571,53 @@ private bool DelegateNeedsReturn(MethodSymbol invokeMethod)
             return true;
         }
 
+        internal NamedTypeSymbol? InferDelegateType(ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            var compilation = Binder.Compilation;
+            var parameterRefKindsBuilder = ArrayBuilder<RefKind>.GetInstance(ParameterCount);
+            var parameterTypesBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(ParameterCount);
+            for (int i = 0; i < ParameterCount; i++)
+            {
+                parameterRefKindsBuilder.Add(HasExplicitlyTypedParameterList ? RefKind(i) : default);
+                parameterTypesBuilder.Add(HasExplicitlyTypedParameterList ? ParameterTypeWithAnnotations(i) : TypeWithAnnotations.Create(compilation.GetSpecialType(SpecialType.System_Object)));
+            }
+            var parameterRefKinds = parameterRefKindsBuilder.ToImmutableAndFree();
+            var parameterTypes = parameterTypesBuilder.ToImmutableAndFree();
+
+            var lambdaSymbol = new LambdaSymbol(
+                Binder,
+                compilation,
+                Binder.ContainingMemberOrLambda!,
+                _unboundLambda,
+                parameterTypes,
+                parameterRefKinds,
+                refKind: default,
+                returnType: default);
+            var lambdaBodyBinder = new ExecutableCodeBinder(_unboundLambda.Syntax, lambdaSymbol, ParameterBinder(lambdaSymbol, Binder));
+            var block = BindLambdaBody(lambdaSymbol, lambdaBodyBinder, BindingDiagnosticBag.Discarded);
+            var returnTypes = ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)>.GetInstance();
+            BoundLambda.BlockReturns.GetReturnTypes(returnTypes, block);
+            var inferredReturnType = BoundLambda.InferReturnType(
+                returnTypes,
+                _unboundLambda,
+                lambdaBodyBinder,
+                delegateType: null,
+                isAsync: IsAsync,
+                Binder.Conversions);
+
+            if (HasExplicitlyTypedParameterList && (inferredReturnType.TypeWithAnnotations.HasType || returnTypes.Count == 0))
```

Comment:
>HasExplicitlyTypedParameterList [](start = 16, length = 31)

What about parameterless lambdas?

Reply:
`HasExplicitlyTypeParameterList` is `true` for lambdas with `() => ...` parameter list.